gio-macros/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+extern crate proc_macro;
+
+mod action_map;
+
+use proc_macro::TokenStream;
+use syn::{parse_macro_input, ItemImpl};
+
+#[proc_macro_attribute]
+pub fn actions(_attr: TokenStream, item: TokenStream) -> TokenStream {
```

Comment:
This could use some documentation with code example(s) :)

Reply:
Yea, sure. I just wanted to share what is done at the moment. Also some tests would be great too.

Reply:
It looks overall good to me, I just can't see how you set the ActionMap, so an example would help figure out how the API is supposed to work :)

Reply:
Seems good to me too apart from the comments above but an example would be nice :)